ip/event_manager.go

```diff
@@ -75,6 +75,10 @@ func (m *manager) eventManager(ctx context.Context) {
 			return
 		}
 
+		interval := time.Duration(m.IP().KeepaliveInterval) * time.Second
```

Comment:
It's not a big computation, but don't you want to compute the interval outside of the loop?

Reply:
ok